### PR TITLE
chore: Some Refactoring

### PR DESCRIPTION
**Pull request by rish691**
Code changes in MetricAnomalyTaskManager, add a new static method to build list of AlertTask 
Some other minor changes related to config and exposing classes

### CODE REVIEW

File: metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/MetricAnomalyAlertTaskJob.java [hunk 1]
```diff
@@ -34,27 +33,39 @@ public void execute(JobExecutionContext jobExecutionContext) throws JobExecution
...
+    List<AlertTask> alertTasks = getAlertTasks(alertTaskConverter, ruleSource);
+    LOGGER.debug("Number of task to execute as part of this run: {}", alertTasks.size());
+    alertTasks.forEach(
+        alertTask -> {
+          try {
+            kafkaAlertTaskProducer.enqueue(alertTask);
+          } catch (IOException e) {
+            LOGGER.debug("Failed execute alert task for task: {} with exception:{}", alertTask, e);
+          }
+          LOGGER.debug("job finished");
```
Comment (kotharironak) [inline metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/job/MetricAnomalyAlertTaskJob.java:45]: This should be outside for each - `job finished`.

File: metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/rule/source/FSRuleSource.java [hunk 2]
```diff
@@ -35,13 +35,15 @@ public FSRuleSource(Config fsConfig) {
       throw new IOException("File should contain an array of notification rules");
     }
 
+    LOGGER.info("Reading alert rules {}", jsonNode.toPrettyString());
```
Comment (kotharironak) [inline metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/rule/source/FSRuleSource.java:38]: Do we need this as info?

File: notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationChannelsReader.java [hunk 3]
```diff
@@ -38,6 +39,8 @@
       throw new IOException("File should contain an array of notification rules");
     }
 
+    LOGGER.info("Reading notification rules {}", jsonNode.toPrettyString());
```
Comment (kotharironak) [inline notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationChannelsReader.java:42]: lets make it to `debug`.

File: notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java [hunk 4]
```diff
@@ -42,14 +42,20 @@
...
     }
-    MetricAnomalyNotificationEvent metricAnomalyNotificationEvent =
-        MetricAnomalyNotificationEvent.fromByteBuffer(eventRecord.getEventValue());
+
+    MetricAnomalyNotificationEvent metricAnomalyNotificationEvent;
+    try {
+      metricAnomalyNotificationEvent =
+          MetricAnomalyNotificationEvent.fromByteBuffer(eventRecord.getEventValue());
+    } catch (IOException e) {
+      throw new RuntimeException("Exception deserializing MetricAnomalyNotificationEvent", e);
```
Comment (kotharironak) [inline notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java:57]: Shouldn't this be IOException only as its serde failure?

**Review by kotharironak — COMMENTED** [review]
overall, lgtm

File: notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationChannelsReader.java (hunk 3 above)
Comment (rish691) [reply notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationChannelsReader.java:42]: this is one time and helpful in knowing what were the rules

File: metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/rule/source/FSRuleSource.java (hunk 2 above)
Comment (rish691) [reply metric-anomaly-task-manager/src/main/java/org/hypertrace/alert/engine/metric/anomaly/task/manager/rule/source/FSRuleSource.java:38]: replied

File: notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java (hunk 4 above)
Comment (rish691) [reply notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java:57]: Wrapping in an unchecked exception

File: notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationChannelsReader.java (hunk 3 above)
Comment (kotharironak) [reply notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationChannelsReader.java:42]: ok, but if it's too much logging, can be wrapped inside debug.

File: notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java (hunk 4 above)
Comment (kotharironak) [reply notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java:57]: But, why?

Comment (kotharironak) [reply notification-service/src/main/java/org/hypertrace/alert/engine/notification/service/NotificationEventProcessor.java:57]: Take care of it in follow-up.

**Review by kotharironak — APPROVED** [review]

**Pull request merged**